Setting screen info from C# code

alfein opened the pull request:
On the core level, I'd like to be able to have the screen object on the javascript side to properly reflect the size and the available size of the monitor hosting the window where browser control lives.

On the Wpf Side, I added an example of how.  

@amaitland, at a minimum there should be a way to propagate this information from the C# callback of GetScreenInfo(), even if at the core level, you chose to not actually implement it, the behavior should be as is.

amaitland commented:
Please review your formatting and revert all the unnecessary changes. 

Spaces not tabs also.

alfein commented:
@amaitland, thanks for checking in.  I fixed formatting of the C++ code and removed all other code from the sample project.    I also added comments to the C++ file.  The changes to the ScreenInfo struct are well-documented.  I don't think any other comments are required.

I think this change is going to be quite useful for anyone willing to add additional C# logic to return monitor information, as currently the 


```javaccript
window.screen
```
object returns wrong values.

amaitland commented:
There are still a huge number of unnecessary formatting changes, please revert. 

Review using the files changed tab up the top of this page.

amaitland commented:
I'd like them reverted thanks.Makes the review easier. I think you've changed files that can be fully reverted at a quick glance.

amaitland commented on CefSharp.Core/Internals/RenderClientAdapter.h:
```diff
@@ -23,7 +23,7 @@ namespace CefSharp
 
         public:
-            RenderClientAdapter(IWebBrowserInternal^ webBrowserInternal, IBrowserAdapter^ browserAdapter):
+            RenderClientAdapter(IWebBrowserInternal^ webBrowserInternal, IBrowserAdapter^ browserAdapter) :
```
This I don't particularly have s problem with, doesn't make the review harder.

amaitland commented on CefSharp.Wpf.Example/MainWindow.xaml.cs:
```diff
@@ -4,12 +4,12 @@
 using CefSharp.Example;
 using CefSharp.Wpf.Example.Controls;
 using CefSharp.Wpf.Example.ViewModels;
+using CefSharp.Wpf.Example.Views;
```
This file looks like it can be fully reverted.

amaitland commented on CefSharp.Wpf/ChromiumWebBrowser.cs:
```diff
@@ -26,7 +26,7 @@
 using Rect = CefSharp.Structs.Rect;
 
-namespace CefSharp.Wpf 
+namespace CefSharp.Wpf
```
Are code changes actually required in this class? Looks like formatting only, this sort of thing makes reviews harder

amaitland replied:
Can this whole file be reverted? Looks like formatting changes only

amaitland commented on CefSharp/Structs/ScreenInfo.cs:
```diff
@@ -7,10 +7,40 @@ namespace CefSharp.Structs
+        /// <summary>
+        /// Available Left location  of the monitor nearest to the window hosting the browser.  iF toolbar is on the left - this would be a non-zero value
+        /// </summary>
+        public int AvailableLeft { get; set; }
```
Use Rect to represent the 4 values https://github.com/cefsharp/CefSharp/blob/master/CefSharp/Structs/Rect.cs

Maybe the AvailableRect property nullable?

amaitland commented on CefSharp/Structs/ScreenInfo.cs:
```diff
@@ -7,10 +7,40 @@ namespace CefSharp.Structs
+        /// <summary>
+        /// Height of the monitor nearest to the window hosting the browser
+        /// </summary>
+        public int Height { get; set; }
```
I think for completeness depth and the other properties should be added.

amaitland commented on CefSharp.Wpf.Example/MainWindow.xaml.cs:
```diff
@@ -4,12 +4,12 @@
 
 using System;
 using System.Collections.ObjectModel;
-using System.Threading.Tasks;
```
Revert this file if there are only formatting changes

alfein commented:
@amaitland, thanks for making changes, but I would have done that over some weekend.  Apologies for being not too responsive.  

Again, thanks for finishing it.  Hope it wasn't too burdensome.

amaitland commented:
> Again, thanks for finishing it. Hope it wasn't too burdensome.

Was very straight forward 😄 

Changes have been merged into `cefsharp/67` branch
